odoo/addons/base/models/ir_model.py

```diff
@@ -767,7 +767,7 @@ def _drop_column(self):
                 if field.state == 'manual' and field.ttype == 'many2many':
                     rel_name = field.relation_table or (is_model and model._fields[field.name].relation)
                     tables_to_drop.add(rel_name)
-            if field.state == 'manual' and is_model:
+            if is_model and (field.state == 'manual' or self._context.get(MODULE_UNINSTALL_FLAG)):
```

Comment:
Read with diff of Strategy3

I am not sure if we should always `_pop_field(field.name)` here, since we don't check `field.state == 'manual'` before `ALTER TABLE %s DROP COLUMN %s CASCADE` and the database and registry can be inconsitstent

even if it looks the most logical solution which tries to keep the consistency between database and registry, it may be risky e.g. we have to supress the error `resolve_depends` below since we cannot cleanup the `registry.field_depends.values()` for removed fields.
And the key error of `model[field_name]` maybe raised  somewhere else.